api/lib/infrastructure/serializers/jsonapi/training-serializer.js

```diff
@@ -9,6 +9,23 @@ module.exports = {
   },
 
   deserialize(payload) {
-    return new Deserializer({ keyForAttribute: 'camelCase' }).deserialize(payload);
+    return new Deserializer({
+      keyForAttribute: 'camelCase',
+      transform(deserializedTraining) {
+        const duration = deserializedTraining.duration;
+        if (!duration) return deserializedTraining;
+
+        const days = duration?.days || 0;
+        const hours = duration?.hours || 0;
+        const minutes = duration?.minutes || 0;
```

Comment:
ça peut être fait dans Joi non ? 🤔

Reply:
je vais regarder ça c'est une bonne idée 👍🏻

Reply:
On préfère le garder, en fait, car dans le cas d'un patch (qui utilise le même serializer), on a pas forcément envie de passer duration avec toutes ses propriétés.
Aussi, on ne voudra pas forcément mettre une valeur par défaut dans Joi pour le patch.

Reply:
Avec le early return `if (!duration)`, il est finalement forcément défini dans la suite.

Au `PATCH`, le `|| 0` risque de reset un des éléments de l'objet si on ne l'envoie pas.

Ex : J'ai un training d'id 1 avec cette duration : `{ days: 4, hours: 4, minutes: 4 }`.

Si je fais cette requête : `PATCH /training/1 { duration: { days: 2 } }`, je me retrouve après deserialization avec cette duration : `{ days: 2, hours: 0, minutes: 0 }` alors que ce n'est probablement pas ce que je voulais.
Je pense que ça serait mieux de forcer la présence de tous les éléments de l'objet si `duration` est présent (voir l'autre commentaire sur le contrat d'interface du PATCH)

Ce qui simplifierai le deserializer :

```suggestion
        const days = duration.days;
        const hours = duration.hours;
        const minutes = duration.minutes;
```